### PR TITLE
mgr/dashboard: Make all columns sortable

### PR DESCRIPTION
**Pull request by Devp00l**
Fixes: https://tracker.ceph.com/issues/39483
Signed-off-by: Stephan Müller <smueller@suse.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.spec.ts [hunk 1]
```diff
@@ -100,6 +100,10 @@ describe('OsdListComponent', () => {
     expect(component).toBeTruthy();
   });
 
+  it('should have columns that are searchable', () => {
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.spec.ts]: "searchable" or "sortable" (as the PR description suggests)?

Comment (Devp00l) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.spec.ts]: done

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts [hunk 2]
```diff
@@ -153,7 +153,7 @@ export class OsdListComponent implements OnInit {
       { prop: 'collectedStates', name: this.i18n('Status'), cellTemplate: this.statusColor },
       { prop: 'stats.numpg', name: this.i18n('PGs') },
       { prop: 'stats.stat_bytes', name: this.i18n('Size'), pipe: this.dimlessBinaryPipe },
-      { name: this.i18n('Usage'), cellTemplate: this.osdUsageTpl },
+      { prop: 'stats.stat_bytes_used', name: this.i18n('Usage'), cellTemplate: this.osdUsageTpl },
```
Comment (s0nea) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts]: I'm wondering if we really want to sort by `stats.stat_bytes_used` and not by the current percentage.  Under the assumption that the size (`stats.stat_bytes`) of the OSDs is not equal to each other, funny things can happen:

Screenshot_2019-03-07_13-37-05

Comment (Devp00l) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts]: Done

**Pull request merged**